src/analysis/asymptotic_equivalent.lean

```diff
@@ -0,0 +1,227 @@
+/-
+Copyright (c) 2020 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker
+-/
+import analysis.asymptotics
+import analysis.normed_space.ordered
+
+/-!
+# Asymptotic equivalence
+
+In this file, we define the relation `is_equivalent u v l`, which means that `u-v` is little o of
+`v` along the filter `l`.
+
+Unlike `is_[oO]` relations, this one requires `u` and `v` to have the same codomaine `β`. While the
+definition only requires `β` to be a `normed_group`, most interesting properties require it to be a
+`normed_field`.
+
+## Notations
+
+We introduce the notation `u ~[l] v := is_equivalent u v l`, which you can use by opening the
+`asymptotics` locale.
+
+## Main results
+
+If `β` is a `normed_group` :
+
+- `_ ~[l] _` is an equivalence relation
+- Equivalent statements for `u ~[l] (λ _, c)` :
+  - If `c ≠ 0`, this is true iff `tendsto u l (𝓝 c)` (see `is_equivalent_const_iff_tendsto`)
+  - For `c = 0`, this is true iff `u =ᶠ[l] 0` (see `is_equivalent_zero_iff_eventually_zero`)
+
+If `β` is a `normed_field` :
+
+- Alternative characterization of the relation (see `is_equivalent_iff_exists_eq_mul`) :
+
+  `u ~[l] v ↔ ∃ (φ : α → β) (hφ : tendsto φ l (𝓝 1)), u =ᶠ[l] φ * v`
+
+- Provided some non-vanishing hypothesis, this can be seen as `u ~[l] v ↔ tendsto (u/v) l (𝓝 1)`
+  (see `is_equivalent_iff_tendsto_one`)
+- For any constant `c`, `u ~[l] v` implies `tendsto u l (𝓝 c) ↔ tendsto v l (𝓝 c)`
+  (see `is_equivalent.tendsto_nhds_iff`)
+- `*` and `/` are compatible with `_ ~[l] _` (see `is_equivalent.mul` and `is_equivalent.div`)
+
+If `β` is a `normed_linear_ordered_field` :
+
+- If `u ~[l] v`, we have `tendsto u l at_top ↔ tendsto v l at_top`
+  (see `is_equivalent.tendsto_at_top_iff`)
+
+-/
+
+namespace asymptotics
+
+open filter
+open_locale topological_space
+
+section normed_group
+
+variables {α β : Type*} [normed_group β]
+
+/-- Two functions `u` and `v` are said to be asymptotically equivalent along a filter `l` when
+    `u x - v x = o(v x)` as x converges along `l`. -/
+def is_equivalent (u v : α → β) (l : filter α) := is_o (u - v) v l
+
+localized "notation u ` ~[`:50 l:50 `] `:0 v:50 := is_equivalent u v l" in asymptotics
+
+variables {u v w : α → β} {l : filter α}
+
+lemma is_equivalent.is_o (h : u ~[l] v) : is_o (u - v) v l := h
+
+lemma is_equivalent.is_O (h : u ~[l] v) : is_O u v l :=
+(is_O.congr_of_sub h.is_O.symm).mp (is_O_refl _ _)
+
+lemma is_equivalent.is_O_symm (h : u ~[l] v) : is_O v u l :=
```

Comment:
If you move it below `is_equivalent.symm`, then you can use `h.symm.is_O` as a proof.

Reply:
Actually that doesn't work since this lemma is used to prove `is_equivalent.is_O_symm`